<?xml version="1.0"?>

<launch>

    <arg name="outfile" default="results"/>
    <!-- gripper, ft sensors and rosserial -->
    <include file="$(find velvet_interface_node)/launch/velvet_ft.launch"/>
    <!-- fri client node -->
    <include file="$(find lbr_fri)/launch/fri_node.launch"/>
    <!-- robot description and rviz -->
    <include file="$(find apple_description)/launch/lbr_velvet_rviz.launch"/>
    <!-- structure io sensor -->
    <!--include file="$(find openni2_launch)/launch/structure_io.launch">
	<arg name="publish_tf" value="false" />
    </include-->

    <include file="$(find experiment_console)/launch/structure_frames.launch"/>
    
    <!-- static publisher for object to world -->
    <node pkg="tf" type="static_transform_publisher" name="obj2world"
	    args="0.9 0.05 0.13 0 0 0  world object_frame 100" />  

    <node pkg="experiment_console" name="console" type="experiment_console_node" launch-prefix="xterm -e ">
	<param name="contact_thresh" value="20" />
	<param name="final_thresh" value="35" />
	<param name="belt_travel" value="100" />
	<param name="phalange_gelta" value="0.02" />
	<param name="gripper_closed_thresh" value="1.5" />
	<param name="check_phalanges" value="true" />
	<param name="outfile_name" value="$(find experiment_console)/output/$(arg outfile)" />

    </node>   

    <node pkg="rosbag" name="recorder" type="record" args="-o $(find experiment_console)/output/$(arg outfile) --split --size 2000 /exp_results" />

</launch>

